sopel/bot.py

```diff
@@ -594,27 +594,26 @@ def rate_limit_info(
     ) -> tuple[bool, Optional[str]]:
         if trigger.admin or rule.is_unblockable():
             return False, None
-
+        nick = trigger.nick
         is_channel = trigger.sender and not trigger.sender.is_nick()
         channel = trigger.sender if is_channel else None
 
-        at_time = trigger.time
-
-        user_metrics = rule.get_user_metrics(trigger.nick)
+        user_metrics = rule.get_user_metrics(nick)
         channel_metrics = rule.get_channel_metrics(channel)
         global_metrics = rule.get_global_metrics()
 
-        if user_metrics.is_limited(at_time - rule.user_rate_limit):
+        at_time = trigger.time
+        if rule.is_user_rate_limited(trigger.nick, at_time):
```

Comment:
```suggestion
        if rule.is_user_rate_limited(nick, at_time):
```

Unrelated to the refactor I'm suggesting re: metrics objects; this just doesn't use the local variable for some reason?